mm2src/common/indexed_db/indexed_cursor.rs

```diff
@@ -0,0 +1,840 @@
+//! # Usage
+//!
+//! As an example, the following table will be used:
+//! ```
+//! | uuid                                   | base_coin | rel_coin | base_coin_value | started_at |
+//! | "c52659d7-4e13-41f5-9c1a-30cc2f646033" | "RICK"    | "MORTY"  | 10              | 1000000029 |
+//! | "5acb0e63-8b26-469e-81df-7dd9e4a9ad15" | "RICK"    | "MORTY"  | 13              | 1000000030 |
+//! | "9db641f5-4300-4527-9fa6-f1c391d42c35" | "RICK"    | "MORTY"  | 1.2             | 1000000031 |
+//! ```
+//! with the `search_index` index created by:
+//! ```rust
+//! TableUpgrader::create_multi_index(self, "search_index", &["base_coin", "rel_coin", "started_at"]).unwrap();
+//! ```
+//!
+//! If you want to find all `RICK/MORTY` swaps where
+//! 1) `10 <= base_coin_value <= 13`
+//! 2) `started_at <= 1000000030`,
+//! you can use [`WithBound::bound`] along with [`WithOnly::only`]:
+//! ```rust
+//! let table = open_table_somehow();
+//! let all_rick_morty_swaps = table.open_cursor("search_index")
+//!     .only("base_coin", "RICK", "MORTY")?
+//!     .bound("base_coin_value", 10, 13)
+//!     .bound("started_at", 1000000030.into(), u32::MAX.into())
+//!     .collect()
+//!     .await?;
+//! ```
+//!
+//! # Under the hood
+//!
+//! In the example above, [`CursorOps::collect`] actually creates a JavaScript cursor with the specified key range:
+//! ```js
+//! var key_range = IDBKeyRange.bound(['RICK', 'MORTY', 10, 1000000030], ['RICK', 'MORTY', 13, 9999999999]);
+//! var cursor = table.index('search_index').openCursor(key_range);
+//! ```
+//!
+//! And after that, the database engine compares each record with the specified min and max bounds sequentially from one field to another.
+//! Please note `['RICK', 'MORTY', 10, 1000000029]` <= `['RICK', 'MORTY', 11, 2000000000]`.
+//!
+//! # Important
+//!
+//! Please make sure all keys of the index are specified
```

Comment:
It seems that created indexes should be represented in the type system somehow to avoid invalid cursor creation. E.g 
```rust
enum SwapIndexes {
    BaseRelStartedAt,
    ...
}
```

It might be a part of the TableSignature trait to pin indexes to a specific table: associated type or another way, just thinking out loud.

Reply:
Thank you for the advice! It makes sense and it's nice to have :)
We can consider integrating it later as you said or if we move it to a public crate